packages/lesswrong/components/editor/CKPostEditor.tsx

```diff
@@ -0,0 +1,194 @@
+import { registerComponent, Components } from '../../lib/vulcan-lib/components';
+import CKEditor from '../editor/ReactCKEditor';
+import { getCkEditor, ckEditorBundleVersion } from '../../lib/wrapCkEditor';
+import { getCKEditorDocumentId, generateTokenRequest} from '../../lib/ckEditorUtils'
+import { CollaborativeEditingAccessLevel, accessLevelCan } from '../../lib/collections/posts/collabEditingPermissions';
+import { ckEditorUploadUrlSetting, ckEditorWebsocketUrlSetting } from '../../lib/publicSettings'
+import { ckEditorUploadUrlOverrideSetting, ckEditorWebsocketUrlOverrideSetting } from '../../lib/instanceSettings';
+import { CollaborationMode } from './EditorTopBar';
+import { defaultEditorPlaceholder } from '../../lib/editor/make_editable';
+import { mentionPluginConfiguration } from "../../lib/editor/mentionsConfig";
+
+// Uncomment this line and the reference below to activate the CKEditor debugger
+// import CKEditorInspector from '@ckeditor/ckeditor5-inspector';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  sidebar: {
+    position: 'absolute',
+    right: -350,
+    width: 300,
+    [theme.breakpoints.down('md')]: {
+      position: 'absolute',
+      right: -100,
+      width: 50
+    },
+    [theme.breakpoints.down('sm')]: {
+      right: 0
+    }
+  },
+})
+
+const refreshDisplayMode = ( editor, sidebarElement ) => {
+  if (!sidebarElement) return null
+  const annotationsUIs = editor.plugins.get( 'AnnotationsUIs' );
+  
+  if ( window.innerWidth < 1000 ) {
+    sidebarElement.classList.remove( 'narrow' );
+    sidebarElement.classList.add( 'hidden' );
+    
+    annotationsUIs.deactivateAll();
+    annotationsUIs.activate('inline');
+  }
+  else if ( window.innerWidth < 1400 ) {
+    sidebarElement.classList.remove( 'hidden' );
+    sidebarElement.classList.add( 'narrow' );
+    
+    annotationsUIs.deactivateAll();
+    annotationsUIs.activate('narrowSidebar');
+  }
+  else {
+    sidebarElement.classList.remove( 'hidden', 'narrow' );
+    
+    annotationsUIs.deactivateAll();
+    annotationsUIs.activate('wideSidebar');
+  }
+}
+
+
+const CKPostEditor = ({ data, collectionName, fieldName, onSave, onChange, documentId, userId, formType, onInit, classes, isCollaborative, accessLevel }: {
+  data?: any,
+  collectionName: CollectionNameString,
+  fieldName: string,
+  onSave?: any,
+  onChange?: any,
+  documentId?: string,
+  userId?: string,
+  formType?: "new"|"edit",
+  onInit?: any,
+  // Whether this is the contents field of a collaboratively-edited post
+  isCollaborative?: boolean,
+  // If this is the contents field of a collaboratively-edited post, the access level the
+  // logged in user has. Otherwise undefined.
+  accessLevel?: CollaborativeEditingAccessLevel,
+  classes: ClassesType,
+}) => {
+  const { EditorTopBar } = Components;
+  const { PostEditor, PostEditorCollaboration } = getCkEditor();
+  const getInitialCollaborationMode = () => {
+    if (!isCollaborative || !accessLevel) return "Editing";
```

Comment:
I agree with this (assuming I understand it).

Reply:
I think this is probably just overly-permissive typing, where `accessLevel` is optional but in reality will always have a value (since it's a resolver-only field that always returns a value).  Need to check that this is the case everywhere that prop is getting passed in.